### PR TITLE
Fix ::User Profile Page - Project/Packages Tabs Have Inconsistent Layout

### PR DESCRIPTION
**Pull request by puni9869**
Fix ::User Profile Page  Project Tab Have Inconsistent Layout and Style
Added the big_avator for consistency in the all header_items tabs.

https://github.com/go-gitea/gitea/issues/24871
> ### Description
> in the user profile page the `Packages` and `Projects` tab have small icons for user but other tabs have bigger profile picture with user info:
> 
> ### Screenshots
> ### **For Packages And Projects:**
> image
> 
> ### **For Other Tabs:**
> image
> 

## Before
image

 ## After changes
Project View 
image

Packages View
image

## Org view for projects page
image

## Org view for packages page
image

### CODE REVIEW

**Comment by puni9869** [comment]
@silverwind

**Comment by silverwind** [comment]
Not sure if related to this PR, but I fixed this issue in https://github.com/go-gitea/gitea/pull/25108/commits/e0813eff1113c76670d1bfef899a3c95b949fe4e:

**Comment by silverwind** [comment]
You missed the `Code` tab which you can see when enabling the repo indexer in `app.ini`:

```ini
[indexer]
REPO_INDEXER_ENABLED = true
REPO_INDEXER_PATH    = indexers/repos.bleve
```

It still renders full width. Please give it the same treatment.

Screenshot 2023-06-06 at 23 18 04

**Comment by puni9869** [comment]
Exactly ,
Can you take it forward. I think `Fix ::User Profile Page - Project/Packages Tabs Have Inconsistent Layout` is done.

**Comment by silverwind** [comment]
@puni9869 Some issues:

1. "Projects" tab is missing while "Code" tab is active: 

Screenshot 2023-06-08 at 11 53 57
Screenshot 2023-06-08 at 11 54 04

2. On "Projects" tab, the tab bar's line is too wide, streching beyond the navbar even:

Screenshot 2023-06-08 at 11 55 11

Code tab search looks alright and does not flash incorrect content, so needs no adjustment imho.

**Comment by silverwind** [comment]
> Also, I see I missed the `all` option in the package selection field, will correct later.

Fixed this in https://github.com/go-gitea/gitea/pull/25108/commits/cc79ebaec5a054a3c58fdf21b52c79c641833ce7.

**Comment by puni9869** [comment]
That headerbar template is shared among all
I'll fix it asap.

**Review by silverwind — APPROVED** [review]
Appears to work fine now.

**Comment by yekanchi** [comment]
looks good to me 2, thatnks

**Comment by puni9869** [comment]
Can we get one more approval.

File: routers/web/shared/user/header.go
Comment (silverwind) [inline routers/web/shared/user/header.go:102]: One more question: Where is this code coming from? Was it copied from somewhere else? Is there a possibility to share at least some of the parts?

Comment (puni9869) [reply routers/web/shared/user/header.go:102]: This is comming from user/profile  where already we are rendering the big_avater and it is used there.

Comment (puni9869) [reply routers/web/shared/user/header.go:102]: Here EnableFeed  --> https://github.com/go-gitea/gitea/blob/main/routers/web/user/profile.go#L51
ShowUserEmail  --> https://github.com/go-gitea/gitea/blob/main/routers/web/user/profile.go#L346


I was unable to test the Badges, How to generate on the profile. But I am confident it will render well.

Comment (puni9869) [reply routers/web/shared/user/header.go:102]: I have made a template that is shared in `template/user/profile` ,  `template/project/list`, `template/user/package`
Shared template https://github.com/go-gitea/gitea/pull/25108/files#diff-6baa9b48bde07abc30f46800ed52c5258d72e2ef73ce33330a489162c6e18f8a

For Project and Package there is shared function https://github.com/go-gitea/gitea/pull/25108/files#diff-2b0b591787f16325539485e648a09ab6d3177f47dc129cfe84a35ffe141dfd19R21

But there is some complexity with user/profile module where also this Big_Avator is used. But Profile function is sharedd between Overview, Activity, Starts  tabs. So I don't want to disterb that flow.

I know some of the render code is duplicate.

Comment (silverwind) [reply routers/web/shared/user/header.go:102]: I guess it's okay if it can not easily be refactored. I'm always on the lookout for duplication because that stuff makes Gitea harder to maintain.

File: routers/web/user/code.go
Comment (wxiaoguang) [inline routers/web/user/code.go:23]: The name and usage seems not ideal.

* The name: it doesn't "render", it only prepares some data
* If the response has been written in it, the caller shouldn't continue (by checking `Written`)

File: templates/projects/list.tmpl
Comment (wxiaoguang) [inline templates/projects/list.tmpl:2]: That's too hacky. Why "IsOrganization"  means "container"?

Comment (silverwind) [reply templates/projects/list.tmpl:2]: If possible, remove the container.

Comment (puni9869) [reply templates/projects/list.tmpl:2]: Unfortunately container cannot be removed, it is hacky beause we are sharing this between repository and orgransation page. I tried removing it but it will mass the whole project tab design in org view.

File: routers/web/user/code.go
Comment (puni9869) [reply routers/web/user/code.go:23]: Correct, and I am even in favour of changing the naming pattern, but when I see https://github.com/go-gitea/gitea/pull/25108/files/6c990c4da74a329f700ffffaa5ddfa1d6195fd7b#diff-f4279417070a8e33829c338abeb42877500377f490abb1495ae6357d50b6a765R92 in the same function doing something like that and implementation looks like this https://github.com/go-gitea/gitea/blob/e24f651c86a5168da8181f2c75f5f99b56561224/routers/web/shared/user/header.go#L13,
 I don't want to break the pattern, if you are suggesting an explicit change in function, that I can.
 
 In short maintainability is the real concern here.
 
Could you suggest something which is easier for maintainers. I have less knowledge on naming conventions.

File: templates/projects/list.tmpl
Comment (silverwind) [reply templates/projects/list.tmpl:2]: Ok, it might be an acceptable compromise then. Didn't check the details.

Comment (wxiaoguang) [reply templates/projects/list.tmpl:2]: My opinion is that these `if-else` tricks are too fragile, so I can't vote my approval, while I won't block if others like it and approve.

File: routers/web/user/code.go
Comment (wxiaoguang) [reply routers/web/user/code.go:23]: Well, the old code is buggy, so I won't say it is a must now.

File: templates/projects/list.tmpl
Comment (puni9869) [reply templates/projects/list.tmpl:2]: Looking for altenates suggestions to take this Pr forward.

**Comment by puni9869** [comment]
There are some major conflicts in this PR. Will re-write this in follow up PR.

**Pull request closed without merging**
